app/javascript/controllers/map_controller.js

```diff
@@ -275,9 +290,11 @@ export default class extends Controller {
     event.preventDefault()
     const tokenId = event.dataTransfer.getData("text/plain")
 
-    const token = this.tokenTargets.find(token => token.dataset.tokenId == tokenId)
+    const token = this.findToken(tokenId)
     if (token.parentNode != this.tokenContainerTarget) {
-      this.tokenContainerTarget.appendChild(token)
+      if (token.dataset.copyOnPlace != "true") {
+        this.tokenContainerTarget.appendChild(token)
+      }
```

Comment:
Why don't we always append the token?

Reply:
When the token is being copied (instead of moved), we don't want to append it (move it from the drawer) because a new token is going to be created and all clients receive it via addToken.

Everything "works" if you don't do this, but there are actually then two tokens on the map, with the exact same position and same `data-token-id`. They move together when you move one, but if you move it fast you can see the other one underneath it.

The reason I added the `count: 1` to the `have_token` matcher was to provide coverage for this bug. It wasn't failing if two of the exact same token appeared on the map.